src/Pos/Explorer/BListener.hs

```diff
@@ -0,0 +1,247 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Blockchain listener for Explorer.
+-- Callbacks on application and rollback.
+
+module Pos.Explorer.BListener
+       ( runExplorerBListener -- (+) BListener instance
+       ) where
+
+import           Control.Monad.Trans.Identity (IdentityT (..))
+import           Data.Coerce                  (coerce)
+import qualified Ether
+
+import           Pos.Block.Types              (Blund)
+import           Pos.DB.BatchOp               (SomeBatchOp)
+import           Pos.Ssc.Class.Helpers        (SscHelpersClass)
+import           Pos.Util.Chrono              (NE, NewestFirst (..),
+                                               OldestFirst (..))
+import           Universum
```

Comment:
Please follow our [style-guide](https://github.com/serokell/serokell-util/blob/master/serokell-style.md#imports) to group imports.